### PR TITLE
Remove remaining unneeded dependencies from Reco subsystems

### PR DESCRIPTION
**Pull request by guitargeek**
#### PR description:

Another quick round of automatic BuildFile.xml cleaning to improve the modularity of CMSSW.

Now, all of the Reco* Build files don't use dependencies anymore which are not included in the source files.

#### PR validation:

CMSSW compiles. It was checked that all newly added dependencies are actually required by the package with `git grep`.

### CODE REVIEW

**Comment by kpedro88** [comment]
please test

**Comment by guitargeek** [comment]
The test got somehow stuck it seems

**Comment by jfernan2** [comment]
please test

**Comment by jfernan2** [comment]
@guitargeek There are DQM changes in HCAL folder for at least wf 11634.0 and 12434.0 
Are they understood or unrelated to this PR?
https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_11_1_X_2020-02-27-2300+88bcf8/35288/dqm-histo-comparison-summary.html

**Comment by guitargeek** [comment]
In this PR, I only changed BuildFiles so there is absolutely no difference expected. The changes must have some other reason I don't know about.

**Comment by jfernan2** [comment]
Since the first test got stuck, I am relaunching it again to see if we reproduce the issue, if you don't mind

**Comment by jfernan2** [comment]
please test

**Comment by kpedro88** [comment]
The changes in DigiTask/Hcal still appear. They all seem to occur in the LETDC subdirectories. @DryRun do you see any reason for this?

**Comment by jfernan2** [comment]
I confirm this is unrelated to this PR but probably to RECO/HCAL feeding DQM: it has just appeared the same difference for a CT-PPS PR in exactly the same workflows for #29037
See: https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_11_1_X_2020-03-02-1100+e25e49/35303/dqm-histo-comparison-summary.html

**Comment by silviodonato** [comment]
> I confirm this is unrelated to this PR but probably to RECO/HCAL feeding DQM: it has just appeared the same difference for a CT-PPS PR in exactly the same workflows for #29037
> See: https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_11_1_X_2020-03-02-1100+e25e49/35303/dqm-histo-comparison-summary.html

Yes, this is a common fake error. See #29076

**Comment by jfernan2** [comment]
+1
Changes are unrelated to this PR

**Comment by kpedro88** [comment]
+upgrade

**Comment by guitargeek** [comment]
Hi @perrotta and @slava77, are you ok with it? I'm just worried that if this PR will be left open much longer, I have to update and test again. This always takes a while because so many depending packages need to be compiled too.

**Comment by slava77** [comment]
@cmsbuild please test

to refresh the tests

File: RecoHI/HiEgammaAlgos/BuildFile.xml [hunk 1]
```diff
@@ -1,9 +1,9 @@
 <use   name="FWCore/Framework"/>
 <use   name="FWCore/ParameterSet"/>
-<use   name="DataFormats/EgammaCandidates"/>
```
Comment (slava77) [inline RecoHI/HiEgammaAlgos/BuildFile.xml]: this was ["conceptually"] a correct dependency.
we do have  in RecoHI/HiEgammaAlgos/interface/TrackIsoCalculator.h
```#include "DataFormats/PatCandidates/interface/Photon.h"```
but the code is actually using ```reco::Photon```
Please revert this change and update RecoHI/HiEgammaAlgos/interface/TrackIsoCalculator.h

File: RecoLocalFastTime/FTLRecProducers/plugins/BuildFile.xml [hunk 2]
```diff
@@ -7,6 +7,7 @@
 <use   name="FWCore/ParameterSet"/>
```
Comment (slava77) [inline RecoLocalFastTime/FTLRecProducers/plugins/BuildFile.xml]: this has a dependency on ```RecoLocalFastTime/FTLRecProducers```
which is technically empty now.
I'm not even sure if  RecoLocalFastTime/FTLRecProducers/BuildFile.xml is even needed.

File: RecoPixelVertexing/PixelTrackFitting/src/KFBasedPixelFitter.cc [hunk 3]
```diff
@@ -29,19 +29,6 @@
 #include "FWCore/Utilities/interface/InputTag.h"
```
Comment (slava77) [inline RecoPixelVertexing/PixelTrackFitting/src/KFBasedPixelFitter.cc]: L12 should be using ```TrackingTools/GeomPropagators/interface/Propagator.h```

File: RecoPixelVertexing/PixelTrackFitting/BuildFile.xml [hunk 4]
```diff
@@ -9,24 +10,21 @@
...
 <use   name="MagneticField/Engine"/>
-<use   name="MagneticField/Records"/>
+<use   name="RecoTracker/TkHitPairs"/>
 <use   name="RecoTracker/TkMSParametrization"/>
 <use   name="RecoTracker/TkTrackingRegions"/>
 <use   name="TrackingTools/DetLayers"/>
-<use   name="TrackingTools/PatternTools"/>
 <use   name="TrackingTools/KalmanUpdators"/>
-<use   name="TrackingTools/Records"/>
+<use   name="TrackingTools/MaterialEffects"/>
```
Comment (slava77) [inline RecoPixelVertexing/PixelTrackFitting/BuildFile.xml]: based on the comment in RecoPixelVertexing/PixelTrackFitting/src/KFBasedPixelFitter.cc, this should not be necessary

**Comment by guitargeek** [comment]
Hi @slava77, thank you very much for checking this and going a bit more in detail than I did.

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/29053	2477826 
- technical updates
- no changes expected; no [relevant] changes observed

**Comment by jfernan2** [comment]
+1

**Comment by pohsun** [comment]
+1

**Comment by slava77** [comment]
@kpedro88 
an upgrade signature is missing.
Please check this PR.
Thank you.

**Comment by kpedro88** [comment]
+upgrade

**Comment by silviodonato** [comment]
+1
